homeassistant/components/tankerkoenig/__init__.py

```diff
@@ -69,87 +79,84 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         return True
 
     conf = config[DOMAIN]
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_NAME: "Home",
+                CONF_API_KEY: conf[CONF_API_KEY],
+                CONF_FUEL_TYPES: conf[CONF_FUEL_TYPES],
+                CONF_LOCATION: {
+                    "latitude": conf.get(CONF_LATITUDE, hass.config.latitude),
+                    "longitude": conf.get(CONF_LONGITUDE, hass.config.longitude),
+                },
+                CONF_RADIUS: conf[CONF_RADIUS],
+                CONF_STATIONS: conf[CONF_STATIONS],
+                CONF_SHOW_ON_MAP: conf[CONF_SHOW_ON_MAP],
+            },
+        )
+    )
 
-    _LOGGER.debug("Setting up integration")
+    return True
 
-    tankerkoenig = TankerkoenigData(hass, conf)
 
-    latitude = conf.get(CONF_LATITUDE, hass.config.latitude)
-    longitude = conf.get(CONF_LONGITUDE, hass.config.longitude)
-    radius = conf[CONF_RADIUS]
-    additional_stations = conf[CONF_STATIONS]
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set a tankerkoenig configuration entry up."""
+    tankerkoenig = TankerkoenigData(hass, entry)
 
-    setup_ok = await hass.async_add_executor_job(
-        tankerkoenig.setup, latitude, longitude, radius, additional_stations
-    )
+    setup_ok = await hass.async_add_executor_job(tankerkoenig.setup)
     if not setup_ok:
         _LOGGER.error("Could not setup integration")
         return False
```

Comment:
Is it always a hard error or could it be that the host can't connect to the server and `ConfigEntryNotReady` is more appropriate?

Reply:
i'm not 100% sure about this, since i'm not very familiar with the underlying lib and the API 🤔 
for now i'll use `ConfigEntryNotReady` , but this needs to be checked further in a separate PR

Reply:
as there is nothing but trash on TV tonight, I found time for this - see 5464e3f17bb7edbd1acdde2428f3ba77abef3b8e